### PR TITLE
Improve Remove Component API (#2368)

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

### What does this PR do?:
- Make cloning the repo optional when removing a component. In some use cases, the repo has already been cloned.  We just want to call the API to only remove the component
- Rename the API so it's more meaningful

Update and more details:

Existing `RemoveAndPush` API has been renamed to `GitRemoveComponent`.  The contents of this function have been refactored out to three different functions/tasks:
- CloneRepo
- RemoveComponent
- CommitAndPush (This was already created in an earlier PR)

Note that this refactoring eliminates the need for extra boolean flags to clone or not to clone, push or not to push, etc.

Alternatively, clients of the library can now call the tasks individually, if desired.

Call cloneRepo first, call RemoveComponent, and then CommitAndPush.

Therefore `GitRemoveComponent` becomes a helper function to do all three tasks.  The Git 'prefix' in the name distinguishes this function from the regular `RemoveComponent` function which doesn't call any git commands.

The new test added reuses the existing test but tests each sub-task individually.

### Which issue(s)/story(ies) does this PR fixes:
2368 

### PR acceptance criteria:
- Test added which is based on the current test for RemoveAndPush

- [x] Unit/Functional tests

- Test added

- [ ] Documentation 

- N/A - current API in code has been updated

- [x] Client Impact

All clients MUST be updated to adopt this change. (I'll do this after)

### How to test changes / Special notes to the reviewer:
Run make test

### CODE REVIEW

File: pkg/gitops.go [hunk 1]
```diff
@@ -283,23 +296,24 @@ func GenerateOverlaysAndPush(outputPath string, clone bool, remote string, optio
...
-	// Checkout the specified branch
-	if _, err := e.Execute(repoPath, "git", "switch", branch); err != nil {
-		if out, err := e.Execute(repoPath, "git", "checkout", "-b", branch); err != nil {
-			return fmt.Errorf("failed to checkout branch %q in %q %q: %s", branch, repoPath, string(out), err)
+	if doClone {
+		if out, err := e.Execute(outputPath, "git", "clone", remote, componentName); err != nil {
+			return fmt.Errorf("failed to clone git repository in %q %q: %s", outputPath, string(out), err)
+		}
+		// Checkout the specified branch
+		if _, err := e.Execute(repoPath, "git", "switch", branch); err != nil {
```
Comment (maysunfaisal) [inline pkg/gitops.go:313]: should this be outside the if condition, what if the repo is already cloned and on a different branch

Comment (keithchong) [reply pkg/gitops.go:313]: Good point. The intent of this is really to not do both the clone and switch.  Just wanted to remove what is present in the filesystem at that time.  So the boolean flag could be renamed to doCloneAndSwitch from doClone.  This is similar to the call to `CommitAndPush` where both the commit and push git commands are run, but the flag is named `doPush`.

Alternatively, we could have yet another bool: doClone and doSwitch to go along with the existing doPush.

Comment (keithchong) [reply pkg/gitops.go:313]: @maysunfaisal , now that we have the refactoring and simplification of types done, and the original APIs are moved over to this repo, I wanted to improve them.  eg. one way is to try to improve the names, starting with this function, so changing the name from RemoveAndPush to simply RemoveComponent.  We will eventually have RemoveEnvironment (or overlay) too.

Comment (keithchong) [reply pkg/gitops.go]: @maysunfaisal , please review this.  I have made the API more cleaner and eliminated the need for the doPush and doClone boolean flags.

**Review by maysunfaisal — APPROVED** [review]

**Comment by keithchong** [comment]
Merged. Thanks @maysunfaisal

**Pull request merged**